v2/tools/generator/internal/config/configurable.go

```diff
@@ -5,11 +5,55 @@
 
 package config
 
+import (
+	"fmt"
+	"github.com/pkg/errors"
+)
+
 // configurable represents a value that may be configured.
 // Includes tracking for whether we consume the configured value or not, allowing us to flag unnecessary configuration
 type configurable[T any] struct {
 	value    *T
 	consumed bool
+	tag      string
+	scope    string
+}
+
+// makeConfigurable creates a new configurable[T] with the given tag and scope
+func makeConfigurable[T any](tag string, scope string) configurable[T] {
+	return configurable[T]{
+		tag:   tag,
+		scope: scope,
+	}
+}
+
+// Lookup returns the value configured, or an error if not configured.
+func (c *configurable[T]) Lookup() (T, error) {
+	if v, ok := c.read(); ok {
+		return v, nil
+	}
+
+	msg := fmt.Sprintf("%s not specified for %s", c.tag, c.scope)
+	return *new(T), NewNotConfiguredError(msg)
+}
+
+// VerifyConsumed returns an error if the value is configured but not consumed.
+func (c *configurable[T]) VerifyConsumed() error {
+	if c.isUnconsumed() {
+		return errors.Errorf("%s specified for %s but not consumed", c.tag, c.scope)
+	}
+
+	return nil
+}
+
+// MarkUnconsumed resets the consumption flag so that the value can be reused
+func (c *configurable[T]) MarkUnconsumed() {
+	c.markUnconsumed()
```

Comment:
Maybe just get rid of the `markUnconsumed` and change all callers to use `MarkUnconsumed`? It seems weird to have this alias here when the actual function is 1 line long.

Reply:
Done.